node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -3747,6 +4009,29 @@ mod tests {
         assert_eq!(dest_db.node_by_key(node_f.public_key()), None);
     }
 
+    fn fix_last_update_nodes(
+        expected_db: &mut NeighborhoodDatabase,
+        dest_db: &NeighborhoodDatabase,
+    ) {
+        let keys = expected_db
+            .keys()
+            .iter()
+            .map(|key| (*key).clone())
+            .collect::<Vec<_>>();
+        keys.into_iter()
```

Comment:
It might be prettier if you write just like this: 

```Rust 
expected_db
  .keys()
  .iter()
  .map(|key| (*key).clone())
  .for_each(|pubkey| match dest_db.node_by_key(&pubkey) {
      Some(node_record) => {
          expected_db
              .node_by_key_mut(&pubkey)
              .unwrap()
              .metadata
              .last_update = node_record.metadata.last_update;
          expected_db.node_by_key_mut(&pubkey).unwrap().resign();
      }
      None => {}
  });
```